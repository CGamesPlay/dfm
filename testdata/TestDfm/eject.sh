#!/bin/bash
# Tests for multiple repositories working correctly
set -e
. "$(dirname "$0")/../helpers.sh"

export HOME="$(pwd)/home"
export DFM_DIR="$HOME/dfmdir"

mkdir -p ~/dfmdir/files
echo 'config' > ~/dfmdir/files/.bashrc
echo 'config' > ~/dfmdir/files/.zshrc

dfm init --repos files
dfm link

banner 'Ejecting one file'
dfm eject ~/.bashrc
rm ~/dfmdir/files/.bashrc
dfm link
[ ! -L ~/.bashrc ] || fail 'bashrc still linked'
[ -e ~/.bashrc ] || fail 'bashrc missing'

banner 'Ejecting everything'
dfm eject
rm ~/dfmdir/files/.zshrc
dfm link
[ ! -L ~/.zshrc ] || fail 'zshrc still linked'
[ -e ~/.zshrc ] || fail 'zshrc missing'
