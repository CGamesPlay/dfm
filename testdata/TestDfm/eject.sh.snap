$ dfm init --repos files
Initialized /test/home/dfmdir as a dfm directory.
$ dfm link
files/.bashrc -> /test/home/.bashrc
files/.zshrc -> /test/home/.zshrc

# Ejecting one file
$ dfm eject /test/home/.bashrc
files/.bashrc -> /test/home/.bashrc
$ dfm link

# Ejecting everything
$ dfm eject
files/.zshrc -> /test/home/.zshrc
$ dfm link
